### PR TITLE
enhancement: Distribute S3 describe_objects

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Enhancement

### Detail
- Distribute S3 describe_objects
- Handle edge case of custom S3 fs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 8d06da766b276e92bd607a1cc04a42aa59f957a0
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6efc46e7-d868-4a34-a150-7280092c80c4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 8d06da766b276e92bd607a1cc04a42aa59f957a0
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f053b4bd-c550-47f0-a594-de2ecfe8877d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 8d06da766b276e92bd607a1cc04a42aa59f957a0
* Result: FAILED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ffa059ab-25d0-4a35-ae46-d8fff665ef19%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: d67e9ed506c353e9e1f88ab1acffb04d4adad50c
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=783aa37d-c5f3-4f40-9091-e042ac672e33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/s3/_describe.py [hunk 1]
```diff
@@ -176,33 +113,28 @@ def describe_objects(
...
+            itertools.repeat(s3_additional_kwargs),
+            [version_id.get(p) if isinstance(version_id, dict) else version_id for p in paths],
+        )
     )
-    size_dict: Dict[str, Optional[int]] = {k: d.get("ContentLength", None) for k, d in desc_list.items()}
-    return size_dict
+    return dict(resp_list)
 
 
 def size_objects(
```
Comment (kukushking) [inline awswrangler/s3/_describe.py:140]: Could you also add kwargs validation, please? Since the related PR was already merged

**Review by kukushking — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: d67e9ed506c353e9e1f88ab1acffb04d4adad50c
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a7f690c9-5c36-4190-a461-d746fcf1bccb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**